core/state/statedb_test.go

```diff
@@ -558,20 +559,6 @@ func TestCopyCommitCopy(t *testing.T) {
 	if val := copyOne.GetCommittedState(addr, skey); val != (common.Hash{}) {
 		t.Fatalf("first copy pre-commit committed storage slot mismatch: have %x, want %x", val, common.Hash{})
 	}
-
-	copyOne.Commit(false)
-	if balance := copyOne.GetBalance(addr); balance.Cmp(big.NewInt(42)) != 0 {
```

Comment:
Why remove this chunk of code?

Reply:
Because once the statedb is committed, then we can't create a copy from it.

The deep reason for this behavior limitation is once the trie(s) are committed, these trie(s) are unusable. We need to re-create statedb to re-initialize all tries with new root upon the updated database.

So yeah, we can't support commit-then-create-a-copy operation.

Reply:
The state db is still usable after commit is: the state objects are still cached in the state db.
The state db is not usable by copying from a committed one is: the cached objects will be discarded and the copied trie is already non-functional.

I will not encourage people to still use the statedb after commit, it just kinda of work, but very hacky.


> I don't think we should just remove the code which tests non-supported order of operations, we should put it somewhere, so we know what happens if the api is misused this way.

I can add a test to show that statedb copied from the committed one is not functional.